dist/sizzle.js

```diff
@@ -1628,13 +1633,237 @@ function setFilters() {}
 setFilters.prototype = Expr.filters = Expr.pseudos;
 Expr.setFilters = new setFilters();
 
-tokenize = Sizzle.tokenize = function( selector, parseOnly ) {
+/**
+ * [AdGuard Patch]:
+ * Sorts the tokens in order to mitigate the performance issues caused by matching slow pseudos first:
+ * https://github.com/AdguardTeam/ExtendedCss/issues/55#issuecomment-364058745
+ */
+var TokenSorter = (function() {
```

Comment:
We don't need a "noun" here. We can make it simply return a `sort` function and save extra property lookup and code size.

Reply:
++